MNIST/main.swift

```diff
@@ -71,48 +71,75 @@ struct Classifier: Layer {
 }
 
 let epochCount = 12
-let batchSize = 100
+let batchSize = 128
 
 func minibatch<Scalar>(in x: Tensor<Scalar>, at index: Int) -> Tensor<Scalar> {
     let start = index * batchSize
     return x[start..<start+batchSize]
 }
 
-let (images, numericLabels) = readMNIST(imagesFile: "train-images-idx3-ubyte",
+let (trainImages, trainNumericLabels) = readMNIST(imagesFile: "train-images-idx3-ubyte",
                                         labelsFile: "train-labels-idx1-ubyte")
-let labels = Tensor<Float>(oneHotAtIndices: numericLabels, depth: 10)
+let trainLabels = Tensor<Float>(oneHotAtIndices: trainNumericLabels, depth: 10)
+
+let (testImages, testNumericLabels) = readMNIST(imagesFile: "t10k-images-idx3-ubyte",
+                                        labelsFile: "t10k-labels-idx1-ubyte")
+let testLabels = Tensor<Float>(oneHotAtIndices: testNumericLabels, depth: 10)
 
 var classifier = Classifier()
-let optimizer = RMSProp(for: classifier)
+
+let optimizer = Adam(for: classifier)
 
 print("Beginning training...")
 
+struct Stats {
```

Comment:
The [Swift API Design Guidelines](https://swift.org/documentation/api-design-guidelines/#use-terminology-well) generally discourages the use of abbreviations in type names unless it's a term of art. Although `Stats` is easily understood, `Statistics` is the term of art here.

```suggestion
struct Statistics {
```